diff --git a/ArduCopter/motors.cpp b/ArduCopter/motors.cpp
index e80894fd6..893c66849 100644
--- a/ArduCopter/motors.cpp
+++ b/ArduCopter/motors.cpp
@@ -32,10 +32,10 @@ void Copter::arm_motors_check()
         return;
     }
 
-    int16_t yaw_in = channel_yaw->get_control_in();
+    int16_t roll_in = channel_roll->get_control_in();
 
     // full right
-    if (yaw_in > 4000) {
+    if (roll_in > 4000) {
 
         // increase the arming counter to a maximum of 1 beyond the auto trim counter
         if (arming_counter <= AUTO_TRIM_DELAY) {
@@ -58,7 +58,7 @@ void Copter::arm_motors_check()
         }
 
     // full left and rudder disarming is enabled
-    } else if ((yaw_in < -4000) && (arming_rudder == AP_Arming::ARMING_RUDDER_ARMDISARM)) {
+    } else if ((roll_in < -4000) && (arming_rudder == AP_Arming::ARMING_RUDDER_ARMDISARM)) {
         if (!flightmode->has_manual_throttle() && !ap.land_complete) {
             arming_counter = 0;
             return;
